pkg/util/result/result.go

```diff
@@ -71,6 +90,38 @@ func FromError(err error) *Result {
 	return &Result{err: err}
 }
 
+// Wrap wraps a Result, producing another Result whose cause is linked to the wrapped Result.
+func Wrap(res *Result, msg string) *Result {
+	return &Result{cause: res, err: errors.New(msg)}
+}
+
+// Wrapf wraps a Result with a formatted message, producing another result whose cause is linked to the wrapped result.
+func Wrapf(res *Result, msg string, args ...interface{}) *Result {
```

Comment:
these comments do not make it clear to me that there will be an error.

Reply:
I think it might be good to just get rid of this for now since it's not being used anywhere yet. We talked about this a few weeks ago and I'm trying to page-in your feedback and I think you recommended something similar about `Cause` and `Wrapf`.